### PR TITLE
Puzzle game with famous IT people

### PR DESCRIPTION
**Pull request by smiechuskce**
## Description
Simple shift-style puzzle game. The game can be started, reset and finished by putting all the tiles in the correct order. For now it was not very secured before JS hacking attempt and the tile-shuffling algorithm can be improved later.

## Where should the reviewer start?
Build app, check files and set test nr 7 opened in DB.

## Motivation and context
#9 

## How has this been tested?

Run app, register or log in and play the game. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Controllers/PuzzleTestController.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> CheckGameStarted()
+        {
+            var user = await _userManager.Users.FirstOrDefaultAsync(u => u.Email == HttpContext.User.Identity.Name);
+
+            if (user != null)
+            {
+                var test = await _context.TestAnswers.FirstOrDefaultAsync(a => a.AnsweringTime == DateTime.MinValue && a.Answer == null && a.UserId == user.Id);
```
Comment (WTobor) [inline src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Controllers/PuzzleTestController.cs]: did you mean _testAnswer_ ? It's better name

**Comment by WTobor** [comment]
We need to wait for changes from #21 to be merged to develop. Then @smiechuskce should make a rebase from _develop_

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Controllers/PuzzleTestController.cs (hunk 1 above)
Comment (smiechuskce) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Controllers/PuzzleTestController.cs]: Changed in layers refactoring commit.

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Models/TestAnswer.cs [hunk 2]
```diff
@@ -15,6 +15,8 @@ public class TestAnswer : ModelBase
         public DateTime AnsweringTime { get; set; }
 
         public TimeSpan AnsweringTimeOffset { get; set; }
+
+        public string Answer { get; set; }
```
[the remark this answers is not in the document]
Comment (smiechuskce) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Models/TestAnswer.cs]: Changed in layers refactoring commit.

**Comment by WTobor** [comment]
Closed due to inactivity

**Pull request closed without merging**